### PR TITLE
Fix changelog tests and add more methods to URLSql

### PR DESCRIPTION
**Pull request by kapillamba4**
fixes https://github.com/short-d/short/issues/200

### CODE REVIEW

File: backend/app/adapter/db/changelog.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ FROM "%s";`,
 		return []entity.Change{}, err
 	}
 
-	var changeLog []entity.Change
+	changeLog := []entity.Change{}
```
Comment (byliuyang) [inline backend/app/adapter/db/changelog.go:36]: `var` automatically initializes the variable with empty array.

File: backend/app/adapter/db/url.go [hunk 2]
```diff
@@ -170,6 +171,101 @@ func (u URLSql) composeParamList(numParams int) string {
 	return parameterStr
 }
 
+// FindURLsByUser finds all urls created by a given user
+func (u URLSql) FindURLsByUser(user entity.User) ([]entity.URL, error) {
```
Comment (byliuyang) [inline backend/app/adapter/db/url.go:175]:
```suggestion
func (u URLSql) FindAliasesByUser(user entity.User) ([]string, error) {
```

File: backend/app/adapter/db/url.go [hunk 3]
```diff
@@ -170,6 +171,101 @@ func (u URLSql) composeParamList(numParams int) string {
...
+	statement := fmt.Sprintf(`
+SELECT %s, %s, %s, %s, %s
+FROM %s
+INNER JOIN %s ON %s.%s = %s.%s
+WHERE "%s"=$1;`,
```
Comment (byliuyang) [inline backend/app/adapter/db/url.go:180]: JOIN statements prevent our service scale to more database shards. Let's avoid using it when possible.

File: backend/app/adapter/db/url.go [hunk 4]
```diff
@@ -170,6 +171,101 @@ func (u URLSql) composeParamList(numParams int) string {
 	return parameterStr
 }
 
+// FindURLsByUser finds all urls created by a given user
```
Comment (byliuyang) [inline backend/app/adapter/db/url.go:174]:
```suggestion
// FindURLsByUser finds the aliases of all the urls created by a given user
```

File: backend/app/adapter/db/url.go [hunk 5]
```diff
@@ -170,6 +171,101 @@ func (u URLSql) composeParamList(numParams int) string {
...
+
+		table.URL.TableName,
+		table.URL.ColumnAlias,
+		table.UserURLRelation.TableName,
+		table.UserURLRelation.ColumnURLAlias,
+
+		table.UserURLRelation.ColumnUserEmail,
+	)
+
+	urls := []entity.URL{}
```
Comment (byliuyang) [inline backend/app/adapter/db/url.go:199]:
```suggestion
	var urls []entity.URL
```

File: backend/app/adapter/db/url.go [hunk 6]
```diff
@@ -170,6 +171,101 @@ func (u URLSql) composeParamList(numParams int) string {
...
+		}
+
+		urls = append(urls, url)
+	}
+
+	return urls, nil
+}
+
+// DeleteURLByUser removes URL having given alias created by given user
+func (u *URLSql) DeleteURLByUser(user entity.User, alias string) (entity.URL, error) {
```
Comment (byliuyang) [inline backend/app/adapter/db/url.go:226]: I wonder whether we are actually using this?

File: backend/app/adapter/db/url.go (hunk 2 above)
Comment (byliuyang) [inline backend/app/adapter/db/url.go:175]: For your information, we have `FindAliasesByUser(user entity.User) ([]string, error)` here: https://github.com/short-d/short/blob/cc1f83f1e1a39be792a5c62888991f75ad0b3ba7/backend/app/adapter/db/userurl.go#L38

File: backend/app/adapter/db/changelog.go (hunk 1 above)
Comment (kapillamba4) [reply backend/app/adapter/db/changelog.go:36]: mdTest failed when i used var

Comment (kapillamba4) [reply backend/app/adapter/db/changelog.go:36]: @byliuyang

File: backend/app/adapter/db/url.go (hunk 2 above)
Comment (kapillamba4) [reply backend/app/adapter/db/url.go:175]: @byliuyang, we need to have longLink also to display table here: https://github.com/short-d/short/pull/615

File: backend/app/adapter/db/url.go (hunk 6 above)
Comment (kapillamba4) [reply backend/app/adapter/db/url.go:226]: Delete button is used in this PR: https://github.com/short-d/short/pull/615

File: backend/app/adapter/db/changelog.go (hunk 1 above)
Comment (kapillamba4) [reply backend/app/adapter/db/changelog.go:36]: Screenshot from 2020-04-04 13-27-00

**Pull request closed without merging**